security/MemberAuthenticator.php

```diff
@@ -51,7 +51,7 @@ protected static function authenticate_member($data, $form, &$success) {
 		if($asDefaultAdmin) {
 			// If logging is as default admin, ensure record is setup correctly
 			$member = Member::default_admin();
-			$success = !$member->isLockedOut() && Security::check_default_admin($email, $data['Password']);
+			$success = Security::check_default_admin($email, $data['Password']) && !$member->isLockedOut();
```

Comment:
I don't think we should change the meeaning of the `$success` var as that will likely have cascading impacts.

I've changed it so `$asDefaultAdmin` won't be true if `Security::has_default_admin` is false. I've also added the `$member` truthy test to the `$success` assignment.